app/controllers/bank_types_controller.rb

```diff
@@ -0,0 +1,53 @@
+class BankTypesController < ApplicationController
+  load_and_authorize_resource
+
+  before_action :load_company
+  before_action :load_data, only: [:index, :new, :create, :edit]
+  before_action :data_bank_type, only: [:create, :update, :destroy]
+
+  def index
+    @bank_types = @company.bank_types
+  end
+
+  def new
+  end
+
+  def create
+    @bank_type = BankType.new bank_type_params
+    if @bank_type.save
+      flash[:notice] = t "flashs.messages.created"
+    end
+  end
+
+  def edit
+  end
+
+  def update
+    if @bank_type.update_attributes bank_type_params
+      flash[:notice] = t "flashs.messages.updated"
+    end
+  end
+
+  def destroy
+    @bank_type.destroy
```

Comment:
nen check before destroy, if it has any data=> redirect_to root with flash : not allow.
If only validate in front-end, not secure